lib/doctest.js

```diff
@@ -77,107 +71,87 @@ module.exports = function(path, options) {
   } else if (options.silent) {
     return Promise.resolve (evaluate (options.module, source, path));
   } else {
-    console.log ('running doctests in ' + path + '...');
-    var results = evaluate (options.module, source, path);
+    console.log (`running doctests in ${path}...`);
+    const results = evaluate (options.module, source, path);
     log (results);
     return Promise.resolve (results);
   }
 };
 
-//  indentN :: (Integer, String) -> String
-function indentN(n, s) {
-  return s.replace (/^(?!$)/gm, (Array (n + 1)).join (' '));
-}
-
-//  object :: Array (Array2 String Any) -> Object
-function object(pairs) {
-  return pairs.reduce (function(object, pair) {
-    object[pair[0]] = pair[1];
-    return object;
-  }, {});
-}
+//    indentN :: (Integer, String) -> String
+const indentN = (n, s) => s.replace (/^(?!$)/gm, ' '.repeat (n));
 
-//  quote :: String -> String
-function quote(s) {
-  return "'" + s.replace (/'/g, "\\'") + "'";
-}
+//    quote :: String -> String
+const quote = s => `'${s.replace (/'/g, "\\'")}'`;
 
-//  show :: a -> String
-function show(x) {
-  return Object.prototype.toString.call (x) === '[object Error]' ?
+//    show :: a -> String
+const show = x =>
+  Object.prototype.toString.call (x) === '[object Error]' ?
     String (x) :
     _show (x);
-}
 
-//  stripLeading :: (Number, String, String) -> String
+//    stripLeading :: (Number, String, String) -> String
 //
-//  > stripLeading (1, '.', 'xxx')
-//  'xxx'
-//  > stripLeading (1, '.', '...xxx...')
-//  '..xxx...'
-//  > stripLeading (Infinity, '.', '...xxx...')
-//  'xxx...'
-function stripLeading(n, c, s) {
-  var idx = 0;
+//    > stripLeading (1, '.', 'xxx')
+//    'xxx'
+//    > stripLeading (1, '.', '...xxx...')
+//    '..xxx...'
+//    > stripLeading (Infinity, '.', '...xxx...')
+//    'xxx...'
+const stripLeading = (n, c, s) => {
+  let idx = 0;
   while (idx < n && s.charAt (idx) === c) idx += 1;
   return s.slice (idx);
-}
+};
 
-//  iifeWrap :: String -> String
-function iifeWrap(s) {
-  return 'void function() {\n' + indentN (2, s) + '}.call(this);';
-}
+//    arrowWrap :: String -> String
+const arrowWrap = s => `void (() => {\n${indentN (2, s)}})();`;
```

Comment:
Do you see any reason not to use an immediately invoked arrow function rather than an immediately invoked function expression?